src/js/base/model.js

```diff
@@ -69,7 +70,30 @@ export default Backbone.Model.extend(extend({
 
     return Backbone.Model.prototype.save.call(this, attrs, opts);
   },
+  set() {
+    const isValid = Backbone.Model.prototype.set.apply(this, arguments);
+
+    if (!isValid) return false;
+
+    this.attributes.__cached_ts = dayjs.utc().format();
+
+    return this;
+  },
   isCached() {
     return this.has('__cached_ts');
   },
+  _getMessageHandler(category) {
+    const messages = result(this, 'messages', {});
+
+    return isFunction(messages[category]) ? messages[category] : this[messages[category]];
+  },
+  handleMessage({ category, timestamp, payload }) {
+    // Ignores messages that may be from recent local events
+    if (dayjs(this.get('__cached_ts')).add(10, 'seconds').isAfter(timestamp)) return;
```

Comment:
@nmajor25 I'm not sure if 10 seconds is a good buffer, but we're trying to account for a slow `PATCH` to the server resulting in a message